### DISCUSSION TITLE
Split schema.prisma into several files

### DISCUSSION THREAD

**Discussion by fnoquiq** [prisma/prisma#5671] [Q&A] [answered]
I would like to split the `schema.prisma` file into two other smaller files, the `user.prisma` and `post.prisma`. How can I do it?

---

Schema.prisma (single file)
```prisma
model User {
  id        String    @id
  name       String
  createdAt  DateTime  @default(now())
  updatedAt  DateTime  @updatedAt
}

model Post {
  id Int @id
  title String
}
```

---

user.prisma (file 1)
```prisma
model User {
  id        String    @id
  name       String
  createdAt  DateTime  @default(now())
  updatedAt  DateTime  @updatedAt
}
```

post.prisma (file 2)
```prisma
model Post {
  id Int @id
  title String
}
```

anothers... (file 3++)

**Comment by johannesschobel** [comment, marked as answer]
this is currently not possible by default prisma tools.
However, you could write your own script, that will do that for you. The code should look like this:

1) define a `global` schema that is used for schema generation, migration and so on. This file should only contain the database-connection parameters and generators needed by default.
2) define a `glob` pattern that finds all `schema` parts. for example, the `glob` may look like this; `**/*.prisma`. This will find all `*.prisma` files within all directories (and subdirectories)
3) from the `glob` pattern, simply `fs.readFile()` the content of all those files and append it to the `global` schema
4) import / load / migrate / generate from the global schema once you `stitched` everything together.

**Reply by nikolasburk** [reply]
Thanks for chiming in @johannesschobel 🙏  I think writing a custom script for this is valid idea! 👌 

@fnoquiq, we also already have an open feature request that you can subscribe to in order to get notified when this is being implemented here: https://github.com/prisma/prisma/issues/2377

**Reply by johannesschobel** [reply]
I think, the _"write your own custom script and use plain FileSystem operations to merge the files"_ is a very valid approach, because it can be achieved in just a few hours ;)
Depending on the application scenario, splitting the schema in various files and merging them back into one big file, may be suitable.

You could even use a `npm script` to automatically merge / generate the global schema file before running `prisma migrate / generate`

**Reply by fnoquiq** [reply]
Thanks for everything, I already subscribe to the topic. I hope they implement soon...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
